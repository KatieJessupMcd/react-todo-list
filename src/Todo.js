import React, { Component } from 'react';

class Todo extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.handleEdit = this.handleEdit.bind(this);
  }

  handleClick() {
    this.props.remove(this.props.id);
  }

  handleEdit() {
    this.props.edit(this.props.id, this.props.task);
  }

  // showEditForm ?

  render() {
    return (
      <div className="to-do-list" id={this.props.id}>
        {' '}
        {this.props.task}
        <button onClick={this.handleClick}>X</button>
        <button onClick={this.handleEdit}>Edit</button>
      </div>
    );
  }
}

export default Todo;
