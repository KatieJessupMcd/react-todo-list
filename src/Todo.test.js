import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Todo from './Todo';

// smoke test
it('renders without crashing', function() {
  shallow(<Todo />);
});

// snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<Todo />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
