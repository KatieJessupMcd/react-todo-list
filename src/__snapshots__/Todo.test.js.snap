// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="to-do-list"
>
   
  <button
    onClick={[Function]}
  >
    X
  </button>
  <button
    onClick={[Function]}
  >
    Edit
  </button>
</div>
`;
